build/build.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+set -o errexit
+set -o nounset
+set -o pipefail
+
+PACKAGE="github.com/openshift/windows-machine-config-operator"
+MAIN_PACKAGE="${PACKAGE}/cmd/manager"
+BIN_NAME="windows-machine-config-operator"
+OUTPUT_DIR="build/_output"
+BIN_DIR="${OUTPUT_DIR}/bin"
+
+echo "building ${BIN_NAME}..."
+mkdir -p "${BIN_DIR}"
```

Comment:
We might want to run `clean` before this

Reply:
Are you suggesting `make clean`? I think checking and deleting the `BIN_DIR` is good enough.

Reply:
I am a nack on this.  This can unnecessarily increase compile time during development. Also in general, you never want to call clean in a build endpoint.

Reply:
It doesn't have to run `make clean` per se - we just have to ensure that `{BIN_DIR}` doesn't already exist, otherwise we will run into errors when we try to run the build targets again without cleaning first

Reply:
Why would we run into errors? Please see the man page for `mkdir -p`.

Reply:
Ohh I misread the code that we've `mkdir -p`. You will not run into errors but it's just a safe guard.

Reply:
It is still unclear if you want the clean to happen before the build.

Reply:
Totally missed the `-p`. Looks good.

Reply:
If you use clean in this script, it would create an inverted dependency, I don't want that.